tsrc/test/cli/test_sync.py

```diff
@@ -2,6 +2,7 @@
 import os
 
 from path import Path
+import pygit2
```

Comment:
Tiny nit - pygit2 is an implementation detail of the GitServer and I'd rather not have leak everywhere.

You should consider adding an `executable` boolean to the `push_file` method instead

Reply:
I thought about that. Since it's specific to the test harness, I don't mind adding an executable kwarg.